### PR TITLE
Fix values in supported being duplicated

### PR DESCRIPTION
**Pull request by Half-Shot**
Pretty straightforward bug. I've added testing to ensure we catch it.

### CODE REVIEW

File: spec/client.spec.ts [hunk 1]
```diff
@@ -13,97 +14,123 @@ describe('Client', () => {
     describe('joining channels', () => {
         test('will get a join event from a newly joined user', async () => {
             const { speaker, listener } = server.clients;
+            const expectedChannel = TestIrcServer.generateUniqueChannel('foobar');
```
Comment (Half-Shot) [inline spec/client.spec.ts:17]: We were previously just using this channel for all tests, which broke when things happened concurrently.

File: src/testing/index.ts [hunk 2]
```diff
@@ -82,12 +85,16 @@ export class TestIrcServer {
                     port: this.port,
                     autoConnect: false,
                     connectionTimeout: 4000,
+                    debug: true,
                 });
             this.clients[clientName] = client;
-            connections.push(new Promise<void>((resolve, reject) => {
+            // Make sure we load isupport before reporting readyness.
+            const isupportEvent = client.waitForEvent('isupport').then(() => undefined);
```
Comment (AndrewFerr) [inline src/testing/index.ts:92]: What's the `then` for?

Comment (Half-Shot) [reply src/testing/index.ts:92]: I was making types work by not returning a value, but it's a bit dirty. Have resolved to just changing the array type.

Comment (AndrewFerr) [reply src/testing/index.ts:92]: Ah, I see.  Didn't realize that `waitForEvent`'s promise held a value.

Then how about `.then(() => {});`?

**Review by AndrewFerr — APPROVED** [review]
LGTM, with one optional request

**Pull request merged**